### DISCUSSION TITLE
chunk size must be greater than 5242880

### DISCUSSION THREAD

**Discussion by chouchouxia** [minio/minio#13387] [Q&A] [answered]
## Expected Behavior
I uploaded chunks in size 2097152(2M), less than 5242880(5M),  then I tried to composed these chunks, I got the following error:

java.lang.IllegalArgumentException: source res/chunk/04374b12a2efef7c2ab53579475525b2/1.chunk: size 2097152 must be greater than 5242880
  at io.minio.MinioClient.calculatePartCount(MinioClient.java:2184) ~[minio-7.1.2.jar!/:7.1.2]
  at io.minio.MinioClient.composeObject(MinioClient.java:2299) ~[minio-7.1.2.jar!/:7.1.2]

Is there any solution to modify the limit that the chunk size must be greater than 5242880?

**Comment by harshavardhana** [comment, marked as answer]
> I uploaded chunks in size 2097152(2M), less than 5242880(5M), then I tried to composed these chunks, I got the following error:

You cannot upload chunks with 2M blocks minimum size required is 5MiB only last part can be < 5MiB - read the S3 spec.

**Comment by G3792274** [comment]
Hi,baby,It's not work.....  @harshavardhana

![image](https://user-images.githubusercontent.com/19231863/146349617-8308425f-6ce1-43eb-a565-e0b06401a34d.png)


![image](https://user-images.githubusercontent.com/19231863/146349362-985c25f7-50c6-4a49-9b75-b82b96e02d03.png)

![image](https://user-images.githubusercontent.com/19231863/146349407-9ae87f9b-d36c-46dd-8e83-eae040447e5b.png)

**Reply by klauspost** [reply]
As said, set a bigger part size. The last part can be < 5MiB, but there is no way the program knows this is the last part.

**Comment by LSun0717** [comment]
> As said, set a bigger part size. The last part can be < 5MiB, but there is no way the program knows this is the last part.

haha
